### ISSUE TITLE
[TF2] enable custom first person animations in Casual

### ISSUE THREAD

**Issue by condor00fr** [ValveSoftware/Source-1-Games#4345]
First person animations should be allowed on Valve servers because the community loves them a lot and they aren't game breaking in any way, besides, disableling them altougether is kinda dumb considering a bug as allowed us players to use them in casual anyways for years, we have proven many times as a community that they don't cause any issues and it would help content creators a lot because as of now only custom animations for the disguise kit, invis watches, the dead ringer and the grappling hooks don't work on casual at all due to the sv_pure1/2 rule.

I think it's time to lift this unessesary and pointess barrier for good. that would be greatly appreciated

**Comment by rabscootle** [comment]
I think they were intentionally disabled due to some animation sets deliberately hiding viewmodels.

**Comment by condor00fr** [comment]
From what I remember Valve stated once that custom viewmodels could give an advantage to theirs users, like less reload time and such. But nah it's not true
Also if you want to hide yout viewmodeld just put your fov to 0. And boom it's gone. Really I can't think of a way as to why it was done, and as stated before. We managed find a way to bipass the sv_pure command years ago, so why keep it up? It just dosn't make any sense anymore.

**Comment by AzureAzel** [comment]
> I think they were intentionally disabled due to some animation sets deliberately hiding viewmodels. Not trying to shut down this ticket, just some insight on why they aren't enabled currently.

I never understood this reasoning. There's already multiple console commands in the game that hide your viewmodels, none of which work in Competitive mode due to their unique game rules. They do all work in Casual, though. I genuinely don't see the problem with allowing custom viewmodel animations in Casual by default if the game already allows you to hide viewmodels. It would be as simple as creating a scenario where custom viewmodel animations are only disabled in Competitive mode by creating a cvar that enables / disables them, and locking the value to disabled on those servers.

It's already possible to preload custom viewmodel animations so they work on Casual servers anyway, and people have been using them for years through this method. Never has it given anyone an unfair advantage at all, at least not anything that can't already be done through currently unrestricted console commands. If it was a problem, I guarantee the TF2 Team would have heard about it already, as it's technically been possible to use them without official support for years.

B4nny and other "iconic" TF2 competitive players already have the console command to hide viewmodels on a toggle with a keybind, which they use all the time in their own competitive matches anyway, so even the notion that we should restrict hiding / moving viewmodels at all in the official competitive mode is questionable at best.

This is why I believe that custom viewmodel animations should be officially supported without having to use a workaround just to use them.

**Comment by Chickenman456** [comment]
> I think they were intentionally disabled due to some animation sets deliberately hiding viewmodels. Not trying to shut down this ticket, just some insight on why they aren't enabled currently.

I see no reason why this is the case as there's already an option to minimize the viewmodels and a console command to remove them entirely.

**Comment by condor00fr** [comment]
Matter in fact, you CAN use custom viewmodel animations in casual. But it requires "preloading" on another server prior to joining matchmaking

A shoot. He said it, now it's gonna get patched next update:(

**Comment by AzureAzel** [comment]
> Here's a workaround.
> 
> Make a CFG file called "autoexec" in your tf/cfg folder.
> 
> paste this
> 
> map itemtest; wait 5; disconnect; wait 3; echo
> 
> Should preload the animations for you. Works for me.

Yep, this is the workaround. But we shouldn't need a workaround just to use them. This could be patched at any time. The point of this thread is to discuss the possibility of making custom animations intentionally supported so we don't need to do that every time when we start the game.

**Comment by Chickenman456** [comment]
It really should be enabled by default.
